go.mod

```diff
@@ -0,0 +1,27 @@
+module sigs.k8s.io/cluster-inventory-api
+
+go 1.21
```

Comment:
is there a reason not to go with latest 1.22?

Reply:
No particular reason. It was the version that the latest stable kubebuilder used so I didn't change it.

I will update this and bring the other dependencies to the latest stable versions as well.

Reply:
Done as suggested. Thanks.